concept_formation/cobweb.py

```diff
@@ -543,24 +551,126 @@ def two_best_children(self, instance):
 
         :param instance: The instance currently being categorized
         :type instance: :ref:`Instance<instance-rep>`
-        :return: the category utility and indices for the two best children (the
-            second tuple will be ``None`` if there is only 1 child).
+        :return: the category utility and indices for the two best children
+            (the second tuple will be ``None`` if there is only 1 child).
         :rtype: ((cu_best1,index_best1),(cu_best2,index_best2))
         """
         if len(self.children) == 0:
             raise Exception("No children!")
 
-        children_cu = [(self.cu_for_insert(child, instance), child.count,
-                        random(), child) for child in self.children]
-        children_cu.sort(reverse=True)
+        children_relative_cu = [(self.relative_cu_for_insert(child, instance),
+                                 child.count, random(), child) for child in
+                                self.children]
+        children_relative_cu.sort(reverse=True)
+
+        # Convert the relative CU's of the two best children into CU scores
+        # that can be compared with the other operations.
+        const = self.compute_relative_CU_const(instance)
+
+        best1 = children_relative_cu[0][3]
+        best1_relative_cu = children_relative_cu[0][0]
+        best1_cu = (best1_relative_cu / (self.count+1) / len(self.children)
+                    + const)
+        best1 = (best1_cu, best1)
+
+        best2 = None
+        if len(children_relative_cu) > 1:
+            best2 = children_relative_cu[1][3]
+            best2_relative_cu = children_relative_cu[1][0]
+            best2_cu = (best2_relative_cu / (self.count+1) / len(self.children)
+                        + const)
+            best2 = (best2_cu, best2)
+
+        return best1, best2
```

Comment:
Modify to return best1_cu, best1, best2 (where best1 and best2 are now pointers to the children not tuples).